### PR TITLE
add tar.gz output containing terraform states for #32

### LINKED ISSUE #32: Output the state to file
Being terraform under the hood, we should respect the need to preserve the state of the VM deployments, or else it will not work smoothly on multi-clients setup

the idea is to output a `tar.gz` of the whole `~/.terrible/$playbook_name` folder to a file called something like `$playbook-name.state.tar.gz` in the same folder of the $playbook file

we should have a `restore` tag that will take care of restoring the files in `~/.terrible/$playbook_name`, but with the `never` tag also, this has to be an **explicit** action to do (to avoid overriding existing states if not wanted)

so we will have for example

```
./inventory-1.yml
./inventory-1.yml.state.tar.gz
```

we will launch

`ansible-playbook -i inventory-multihyper.yml -u root main.yml --tags restore`

this will populate the folders in `~/.terrible`

then we can proceed with a deployment that will have a known state

### PR DESCRIPTION
**Pull request by 89luca89**
Fix #32

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -547,6 +547,15 @@ To completely delete the infrastructure:
 ansible-playbook -i inventory.yml -u root main.yml --tags purge
 ```
 
+### Outputs
+
+Based on your inventory, the complete state of the infrastructure (TF files, TF states, LUKS keys etc...)
+will be outputted to `${INVENTORY_NAME}-state.tar.gz` this file is important to track the
```
Comment (alegrey91) [inline README.md:553]: better use "recorded" instead of "outputted"? What do you mean?

File: roles/terrible/tasks/assert_vars.yml [hunk 2]
```diff
@@ -294,7 +294,7 @@
   assert:
     quiet: yes
     that:
-      - "{{ item }}.encryption == True or {{ item }}.encryption == False"
+      - "{{ item }}.encryption or not {{ item }}.encryption"
```
Comment (alegrey91) [inline roles/terrible/tasks/assert_vars.yml:297]: Did you test this statement? The reason I used the previous notation was that using your, I was not able to trigger the assert.

**Review by alegrey91 — COMMENTED** [review]
The pr seems to be nice, but please, take notes about my comments.

File: roles/terrible/tasks/assert_vars.yml (hunk 2 above)
Comment (89luca89) [reply roles/terrible/tasks/assert_vars.yml]: Now it's fixed without doing literals

File: README.md (hunk 1 above)
Comment (89luca89) [reply README.md:553]: Yep seems better, Fixed

File: roles/terrible/tasks/assert_vars.yml [hunk 3]
```diff
@@ -294,7 +294,7 @@
   assert:
     quiet: yes
     that:
-      - "{{ item }}.encryption or not {{ item }}.encryption"
+      - "{{ item.encryption }} | type_debug == 'bool'"
```
Comment (alegrey91) [inline roles/terrible/tasks/assert_vars.yml:297]: now looks better!

**Review by alegrey91 — COMMENTED** [review]
The pr is now ready to be merged

**Pull request merged**